pymc/model_graph.py

```diff
@@ -357,26 +436,35 @@ def make_graph(
         node_formatters = update_node_formatters(node_formatters)
 
         graph = graphviz.Digraph(self.model.name)
-        for plate_label, all_var_names in self.get_plates(var_names).items():
-            if plate_label:
+        for plate in self.get_plates(var_names):
+            plate_meta = plate.meta
+            all_vars = plate.variables
+            if plate_meta.names or plate_meta.sizes:
```

Comment:
Can we simplify? Could plate_meta be None for the scalar variables?

Reply:
This logic would still be needed somewhere. Likely in `get_plates` then. 
How about having the __bool__ method for `Plate` class that does this logic. 
Then would act like None and read like: 
```python
if plate_meta: # Truthy if sizes or names
    # plate_meta has sizes or names that are not empty tuples
```

Reply:
You have that information when you defined the plate_meta no? Can't you do it immediately?

Reply:
Yes. I changed to have it happen in the `get_plates` methods. Scalars will have Plate(meta=None, variables=[...])

Reply:
I think it's enough to check for `sizes`? It is not possible for a plate to have names, but not sizes?

We should rename those to `dim_names, and dim_lengths`. And perhaps use `None` for dim_lengths for which we don't know the name?

Reply:
So IIUC, scalars should belong to a "Plate" with dim_names = (), and dim_lengths = ()?

Reply:
And now I understand your approach and I think it was better like you did. The __bool__ sounds fine as well!

Sorry I got confused by the names of the things